WalletWasabi/Models/UpdateStatus.cs

```diff
@@ -21,9 +23,9 @@ public UpdateStatus(bool backendCompatible, bool clientUpToDate)
 
 		public bool Equals(UpdateStatus other) => this == other;
 
-		public override int GetHashCode() => ClientUpToDate.GetHashCode() ^ BackendCompatible.GetHashCode();
+		public override int GetHashCode() => ClientUpToDate.GetHashCode() ^ BackendCompatible.GetHashCode() ^ LegalDocumentsVersion.GetHashCode();
```

Comment:
This is wrong because `ClientUpToDate` and `BackendCompatible` are booleans and the `GetHashCode` of booleans are: 

* true.GetHashCode() = 1
* false.GetHashCode() = 0

So, there are more than one `UpdateStatus` with different values that will return the exact same hash code. For example: 

```c#
var ver = new Version(1, 7, 9);

var hc1 == true.GetHashCode() ^ false.GetHashCode() ^ ver.GetHashCode();
var hc2 == false.GetHashCode() ^ true.GetHashCode() ^ ver.GetHashCode();
hc1 == hc2 // true

var hc3 == true.GetHashCode() ^ true.GetHashCode() ^ ver.GetHashCode();
var hc4 == false.GetHashCode() ^ false.GetHashCode() ^ ver.GetHashCode();

hc3 == hc4 // true
```
```suggestion
		public override int GetHashCode() => (13 * ClientUpToDate.GetHashCode()) ^ ( 47 * BackendCompatible.GetHashCode()) ^ LegalDocumentsVersion.GetHashCode();
```

Reply:
ACK. I didn't know that. I opened an issue, as I may had made this mistake in other places before: https://github.com/zkSNACKs/WalletWasabi/issues/2973

Reply:
Also fixed.